runtime/drivers/admin.go

```diff
@@ -3,14 +3,21 @@ package drivers
 import (
 	"context"
 	"time"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
 )
 
 type AdminService interface {
-	GetReportMetadata(ctx context.Context, reportName string, annotations map[string]string, executionTime time.Time) (*ReportMetadata, error)
+	GetReportMetadata(ctx context.Context, reportName string, reportSpec *runtimev1.ReportSpec, executionTime time.Time) (*ReportMetadata, error)
```

Comment:
Could it extract the list of recipients on the caller side, and just pass that here? Just to avoid having report spec handling logic spread too many places.

Reply:
Its also used to get web open state and title, so we may need to pass those as well. In future if we want owner attr as part of this response then we may need to pass that as well. Let me know if you still want to change to just these 3 parameters instead of report spec.

Reply:
Hmm, maybe I'm misunderstanding, I don't see it accessing open state and title here? https://github.com/rilldata/rill/blob/61d69b09e0b739c53c1f6496a236ada771eb0cb1/admin/server/reports.go#L30

The web open state should just be retrieved by the frontend when looking up the report in the [report open route](https://github.com/rilldata/rill/blob/61d69b09e0b739c53c1f6496a236ada771eb0cb1/admin/urls.go#L308). The title I guess is needed for the email, but that's sent on the runtime side, which already has the report spec, right?

Reply:
I was setting up the state in the magic token for external recipients but yeah front end can directly fetch report resource with the jwt to get the state